src/state/events.cpp

```diff
@@ -0,0 +1,43 @@
+#include "app.hpp"
+
+#define G_LOG_DOMAIN "genie::state::events"
+
+namespace genie {
+namespace state {
+namespace events {
+
+// ### Conversation Events ###
+
+InputFrame::InputFrame(AudioFrame *frame) : frame(frame) {}
+InputFrame::~InputFrame() {
+  // NOTE We do **NOT** free the frame on destruction -- it's handed off to the
+  //      STT instance, where it may be queued for later transmission.
+}
+
+TextMessage::TextMessage(const gchar *text) : text(g_strdup(text)) {}
```

Comment:
These constructors are tiny and are better declared in the class itself (so they become inline)

Reply:
I'll move 'em there

Reply:
Fixed in 1dd6d2cac536547cdfcc4ec918eac1ef8651b5b1